core/modules/block/block.module

```diff
@@ -279,6 +279,19 @@ function block_custom_block_save(array $edit, $delta = NULL) {
   $config->setData($block);
   $config->save();
 
+  // Save each file as permanent, preventing it from being deleted.
+  $fids = filter_parse_file_fids($block['body']['value']);
+  $files = file_load_multiple($fids);
+  foreach ($files as $fid => $file) {
+    if ($file && $file->status !== FILE_STATUS_PERMANENT) {
+      $file->status = FILE_STATUS_PERMANENT;
+      $file->save();
+    }
+    // Essentially this claims that the file is "self-referencing", which means
+    // it will never be deleted until a file management tool is available.
+    file_usage_add($file, 'file', 'file', $file->fid);
+  }
+
```

Comment:
Isn't this duplication here? These blocks can't really be used except in a layout, which means on layout save, this process will happen twice.